alica_engine/include/engine/RunnableObject.h

```diff
@@ -81,7 +102,7 @@ class RunnableObject
     // If the counter is even then it indicates the behaviour is active i.e it is started, but not stopped yet
     constexpr static bool isActive(Counter cnt) { return !(cnt & 1); };
     ThreadSafePlanInterface getPlanContext() const;
-    void setTerminatedState();
+    bool setTerminatedState();
```

Comment:
please document what is this return bool

Reply:
Some context: This method is not very intuitive, infact I had refactored the entire code in RunnableObject, BasicBehavoiur & BasicPlan in https://github.com/rapyuta-robotics/alica/pull/203, but it didn't go through & instead we opted to do a complete redesign https://www.wrike.com/open.htm?id=808599830

So this PR is intended just to pick the changes from release with bare minimum changes required to fix the bug mentioned in the commit